### PR TITLE
fix: change the expectedType of the kube-client reflector to match the listWatch

### LINKED ISSUE #719: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

Deployed `quay.io/coreos/kube-state-metrics:v1.6.0-rc.0`

I keep getting these messages in logs, over and over:

```
E0410 15:22:20.836131       1 reflector.go:329] k8s.io/kube-state-metrics/internal/collector/builder.go:482: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
E0410 15:22:20.842987       1 reflector.go:329] k8s.io/kube-state-metrics/internal/collector/builder.go:482: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
E0410 15:22:20.876726       1 reflector.go:329] k8s.io/kube-state-metrics/internal/collector/builder.go:482: expected type *v1beta1.Deployment, but watch event object had type *v1.Deployment
```

**What you expected to happen**:

It shouldn't log errors.

**How to reproduce it (as minimally and precisely as possible)**:

This my deployment yaml file:

[kube-state-metrics.yaml.txt](https://github.com/kubernetes/kube-state-metrics/files/3064561/kube-state-metrics.yaml.txt)

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.13.3
- Kube-state-metrics image version v1.6.0-rc.0

### PR DESCRIPTION
**Pull request by tariq1890**
Fixes #719 

This PR adds what was missed in PR #652.

### CODE REVIEW

**Comment by tariq1890** [comment]
This needs to be merged into the `release-1.6` branch.

**Comment by brancz** [comment]
Do I understand correctly, that with this we're getting both appsv1 and backward compatible extensionsv1beta1 support back?

cc @mxinden

**Comment by mxinden** [comment]
> This needs to be merged into the release-1.6 branch.

In that case this pull request needs to point to the `release-1.6` branch, @tariq1890 . Once merged, we can merge `release-1.6` back into _master_, preserving git commit hashes across branches.

**Comment by tariq1890** [comment]
@brancz I'll have to test with a pod created with that api Group and see if it's captures and represented properly when kube state metric queries the pod. I haven't had much time on my hands off late. Will provide an update on this as soon as I get the bandwidth.

**Comment by brancz** [comment]
I agree with @LiliC let's just use v1apps for Deployments, StatefulSets and DaemonSets. It'll also be future proof.

**Comment by tariq1890** [comment]
Do you want me to do that in this PR @brancz or in another one? Currently this PR is scoped to fixing #719. But that's just me saying that :).

**Comment by brancz** [comment]
The way I see it is this would be future proofing _and_ fixing the bug :)

**Comment by tariq1890** [comment]
I just tested a deployment manifest created with `extensions/v1beta1` apiGroup with the kube-state-metrics code and I could see the metrics being exported successfully. I think it's safe to say that backwards compatibility with the deprecated `extensions/v1beta1` apiGroup is preserved.

I will start implementing @LiliC's suggestion now.

File: internal/collector/statefulset.go [hunk 1]
```diff
@@ -111,25 +111,21 @@ var (
...
-				if s.Status.ObservedGeneration != nil {
-					ms = append(ms, &metric.Metric{
-						Value: float64(*s.Status.ObservedGeneration),
-					})
-				}
-
+			GenerateFunc: wrapStatefulSetFunc(func(s *v1.StatefulSet) *metric.Family {
 				return &metric.Family{
-					Metrics: ms,
+					Metrics: []*metric.Metric{
```
Comment (tariq1890) [inline internal/collector/statefulset.go]: This change is merely a simplification.

**Comment by tariq1890** [comment]
CI passed and ready for review.

**Comment by mxinden** [comment]
Would you mind changing the target branch to _release-1.6_? See https://github.com/kubernetes/kube-state-metrics/pull/720#issuecomment-483182211.

File: vendor/modules.txt [hunk 2]
```diff
@@ -117,14 +117,14 @@ gopkg.in/inf.v0
...
 k8s.io/api/apps/v1
-k8s.io/api/apps/v1beta1
 k8s.io/api/autoscaling/v2beta1
 k8s.io/api/batch/v1
 k8s.io/api/batch/v1beta1
 k8s.io/api/certificates/v1beta1
 k8s.io/api/core/v1
 k8s.io/api/extensions/v1beta1
 k8s.io/api/policy/v1beta1
+k8s.io/api/apps/v1beta1
```
Comment (LiliC) [inline vendor/modules.txt]: Just curious why go mods do this change, moves the import in this file. The pkg is not used anymore in the code directly, guessing its used by one of the other deps e.g. client-go internally, but it makes me wonder if we can bump the version of client-go and in turn get rid of these dependencies. (But that might be going down the rabbit hole of solving deps, and probably falls outside of this PR :)

**Comment by tariq1890** [comment]
@mxinden Done!

**Comment by brancz** [comment]
/lgtm

but giving @mxinden the last call

/hold

File: vendor/modules.txt (hunk 2 above)
Comment (tariq1890) [reply vendor/modules.txt]: That makes sense. But yes, this definitely is out of scope for this PR. I believe we have k8s 1.14 depped into kube state metrics. We are just behind the latest by a patch release.

**Comment by mxinden** [comment]
Thanks for the fix @tariq1890!

/hold cancel

**Comment by AndrewDryga** [comment]
This issue has a regression, v1.9.6:
```
"pkg/mod/k8s.io/client-go@v0.0.0-20191109102209-3c0d1af94be5/tools/cache/reflector.go:108: expected type *v1.MutatingWebhookConfiguration, but watch event object had type *v1beta1.MutatingWebhookConfiguration"
```

**Comment by lizhifengones** [comment]
> > expected type *v1.MutatingWebhookConfiguration, but watch event object had type *v1beta1.MutatingWebhookConfiguration"
> 
> Is there any response?

#1142

**Pull request merged**